backend/infrahub/tasks/registry.py

```diff
@@ -42,21 +42,31 @@ async def create_branch_registry(db: InfrahubDatabase, branch: Branch) -> None:
 
 
 async def update_branch_registry(db: InfrahubDatabase, branch: Branch) -> None:
-    """Update the registry for a branch if the schema hash has changed."""
+    """Update the registry for a branch if the schema hash has changed or the branch was rebased."""
 
     existing_branch: Branch = registry.branch[branch.name]
 
     if not existing_branch.schema_hash:
         log.warning("Branch schema hash is not set, cannot update branch registry")
         return
 
-    if existing_branch.schema_hash and existing_branch.schema_hash.main == branch.active_schema_hash.main:
+    if existing_branch.schema_hash.main == branch.active_schema_hash.main:
```

Comment:
I removed the first part of the if statement here for the conditional `existing_branch.schema_hash` as we already have a check for this just above and return early if the schema hash is not set.